runelite-client/src/main/java/net/runelite/client/plugins/microbot/qualityoflife/QoLPlugin.java

```diff
@@ -319,83 +425,147 @@ private void customAnvilOnClicked(MenuEntry event) {
-            return;
-        }
-
-        Microbot.getClient().createMenuEntry(entries.size())
+        int index = Microbot.getClient().getMenuEntries().length;
+        Microbot.getClient().createMenuEntry(index)
                 .setOption(option)
                 .setTarget(target)
                 .setParam0(event.getActionParam0())
                 .setParam1(event.getActionParam1())
                 .setIdentifier(event.getIdentifier())
                 .setType(event.getMenuEntry().getType())
                 .onClick(callback);
-
     }
 
     private void addLoadoutMenuEntry(MenuEntryAdded event, String option, String target, Consumer<MenuEntry> callback) {
-        Microbot.getClient().createMenuEntry(1)
+        int index = Microbot.getClient().getMenuEntries().length - 1;
+        Microbot.getClient().createMenuEntry(index)
                 .setOption(option)
                 .setTarget(target)
                 .setParam0(event.getActionParam0())
                 .setParam1(event.getActionParam1())
                 .setIdentifier(event.getIdentifier())
                 .setType(event.getMenuEntry().getType())
                 .onClick(callback);
+    }
+
+    public void updateLastWinthertodtAction(WintertodtActions action) {
+        configManager.setConfiguration("QoL", "wintertodtActions", action);
+    }
+
+    public void updateWintertodtInterupted(boolean interupted) {
+        configManager.setConfiguration("QoL", "interrupted", interupted);
+    }
+
 
+    private void updateUiElements() {
+        try {
+            // Get the Field object for BRAND_ORANGE(Accent color)
+            Field accentColorField = ColorScheme.class.getDeclaredField("BRAND_ORANGE");
+            Class<?> pluginButton = Class.forName("net.runelite.client.plugins.config.PluginToggleButton");
+            Field onSwitcherPluginPanel = pluginButton.getDeclaredField("ON_SWITCHER");
+            onSwitcherPluginPanel.setAccessible(true);
+
+
+            FieldUtil.setFinalStatic(accentColorField, config.accentColor());
+
+            FieldUtil.setFinalStatic(onSwitcherPluginPanel, remapImage(SWITCHER_ON_IMG, config.toggleButtonColor()));
+            ConfigPlugin configPlugin = (ConfigPlugin) Microbot.getPluginManager().getPlugins().stream().filter((plugin) ->
+                    plugin instanceof ConfigPlugin).findAny().orElse(null);
+            if (configPlugin == null) {
+                throw new RuntimeException("Config Plugin not found");
+            } else {
+                try {
+                    Class<?> pluginListPanelClass = Class.forName("net.runelite.client.plugins.config.PluginListPanel");
+                    JPanel pluginListPanel = (JPanel) configPlugin.getInjector().getProvider(pluginListPanelClass).get();
+                    java.util.List<?> pluginList = (java.util.List<?>) FieldUtils.readDeclaredField(pluginListPanel, "pluginList", true);
+
+                    if (pluginList == null) {
+                        sleep(8000);
```

Comment:
why are we sleeping 8 seconds here?

Reply:
the pluginList is still null while the client is starting and initializing all plugins the 8 seconds is just to avoid error spamming and allow the client to finish. Do you want me to modify it to "sleepUntil"?

Reply:
Yeah sleepuntil makes more sense